components/brave_new_tab_ui/stories/default/data/storybookState.ts

```diff
@@ -55,51 +60,100 @@ function getWidgetStackOrder (firstWidget: string): NewTab.StackWidget[] {
+    ),
+    backgroundWallpaper: select(
+      'Background image',
+      generateStaticImages(images),
+      generateStaticImages(images).SpaceX
+    ),
+    customLinksEnabled: boolean('CustomLinks Enabled?', false),
+    featureFlagBraveNewsEnabled: true,
+    featureFlagBraveNewsPromptEnabled: true,
+    searchPromotionEnabled: false,
+    forceSettingsTab: select('Open settings tab?', [undefined, ...Object.keys(SettingsTabType)], undefined),
+    showBackgroundImage: boolean('Show background image?', true),
+    showStats: boolean('Show stats?', true),
+    showToday: boolean('Show Brave News?', true),
+    showClock: boolean('Show clock?', true),
+    showTopSites: boolean('Show top sites?', true),
+    showRewards: boolean('Show rewards?', true),
+    showBraveTalk: boolean('Show Brave Talk?', true),
+    braveTalkSupported: boolean('Brave Talk supported?', true),
+    braveTalkPromptDismissed: !boolean('Brave Talk prompt?', false),
+    geminiSupported: boolean('Gemini Supported?', true),
+    cryptoDotComSupported: boolean('Crypto.com supported?', true),
+    ftxSupported: boolean('FTX supported?', true),
+    showFTX: boolean('Show FTX?', true),
+    showBinance: boolean('Show Binance?', true),
+    hideAllWidgets: boolean('Hide all widgets?', false),
+    isBraveTodayOptedIn: boolean('Brave Today opted-in?', false),
+    textDirection: select('Text direction', { ltr: 'ltr', rtl: 'rtl' }, 'ltr'),
+    stats: {
+      ...state.stats,
+      adsBlockedStat: number('Number of blocked items', 1337),
+      httpsUpgradesStat: number('Number of HTTPS upgrades', 1337)
+    },
+    // TODO(petemill): Support binance state when binance can be included without chrome.* APIs
+    // binanceState: {
+    //   ...state.binanceState,
+    //   binanceSupported: boolean('Binance supported?', true)
+    // },
+    initialDataLoaded: true,
+    widgetStackOrder: getWidgetStackOrder(select('First widget', ['braveTalk', 'rewards'], 'rewards'))
+  }
+
+  // On all updates, notify that the prefs might've changed. Listeners are
+  // only notified when the setting they're interested in updates anyway.
+  useEffect(() => {
+    newTabPrefManager.notifyListeners(result)
+  })
+
+  // In Storybook, we aren't wired up to the settings backend. This effect
+  // **VERY** hackily overrides the savePref function to emit a knob change
+  // event, guessing what the correct knob label is.
+
+  // This is unfortunate, but the `addon-knob` library is deprecated and does
+  // not offer any better way of setting knob values.
+
+  // See https://github.com/storybookjs/storybook/issues/3855#issuecomment-624164453
+  // for discussion of this issue.
+  useEffect(() => {
+    const originalSavePref = newTabPrefManager.savePref
+    newTabPrefManager.savePref = (key, value) => {
+      addonsChannel.emit(CHANGE, {
+        name: guessLabelForKey(key),
+        value: value
+      })
+    }
+    return () => {
+      newTabPrefManager.savePref = originalSavePref
+    }
```

Comment:
When does this get used? When I use the [latest storybook from this PR](https://brave-jenkins-build-artifacts.s3.amazonaws.com/brave-browser-build-pr/ntp-pref-hooks/fa98dbeb7400c3765d8f5e2fc8d54c4d79c3222d/storybook/index.html?path=/story/new-tab--regular), I don't see that any of the UI that sets `showStats` or `showTopSites` gets persisted. To be honest, I don't really think that's important anyway, since the storybook toggles work fine.

Reply:
It's not persisted, this just let's us control the storybook knob from the component. It's not too necessary though, as none of the other stuff is interactive either.

[Screencast from 18-08-22 12:56:37.webm](https://user-images.githubusercontent.com/7678024/185269219-7852f8e8-8bf4-4cde-853a-18b76df12cd6.webm)

Happy to remove it if you don't think we need it :smile:

Reply:
It basically mocks out the PrefHook backend with the StoryBook knob

Reply:
OK yeah I see which one it's working on now. I think it's ok to keep. Not sure how important keep interactivity working like this in Storybook is, but I guess it's nice to have!